core/src/subgraph/registrar.rs

```diff
@@ -575,8 +575,17 @@ async fn create_subgraph_version<C: Blockchain, S: SubgraphStore>(
     .map_err(SubgraphRegistrarError::ResolveError)
     .await?;
 
+    // Determine if the subgraph should be validated.
+    // Validate the subgraph if there is a pending graft, indicating the presence of a graft base.
+    // If the subgraph is new (DeploymentNotFound), it should also be validated.
+    // If the subgraph already exists and there is no pending graft, validation is not required.
```

Comment:
The wording here is not very precise, this boolean is not deciding if the whole subgraph should be validated, but if the graft base should be validated.

Reply:
// Determine if the graft_base should be validated.
// Validate the graft_base if there is a pending graft, ensuring its presence.
// If the subgraph is new (indicated by DeploymentNotFound), the graft_base should be validated.
// If the subgraph already exists and there is no pending graft, graft_base validation is not required.


@leoyvens how does this look?